From roel.kluin@gmail.com  Wed Feb  4 15:03:19 2009
From: Roel Kluin <roel.kluin@gmail.com>
Date: Sat, 31 Jan 2009 12:37:04 +0100
Subject: USB: count reaches -1, tested 0
To: dbrownell@users.sourceforge.net, gregkh@suse.de
Message-ID: <498437E0.3020303@gmail.com>


With a postfix decrement count will reach -1 rather than 0,
so the warning will not be issued.

Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/pci-quirks.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/host/pci-quirks.c
+++ b/drivers/usb/host/pci-quirks.c
@@ -234,7 +234,7 @@ static void __devinit quirk_usb_disable_
 	 */
 	hcc_params = readl(base + EHCI_HCC_PARAMS);
 	offset = (hcc_params >> 8) & 0xff;
-	while (offset && count--) {
+	while (offset && --count) {
 		u32		cap;
 		int		msec;
 
